bootstrap/src/basis/ordmap.ml

```diff
@@ -1671,6 +1689,31 @@ let%expect_test "of_alist,to_alist,to_array" =
     of_alist [(0, "0"); (1, "1"); (66, "66"); (91, "91")]; to_alist -> [(0, "0"); (1, "1"); (66, "66"); (91, "91")]; to_array -> [|(0, "0"); (1, "1"); (66, "66"); (91, "91")|]
     |}]
 
+let%expect_test "choose_hlt" =
+  let open Format in
+  printf "@[";
+  (* test is n^2 time complexity, so keep n small. *)
+  let rec test n i ordmap = begin
+    match i < n with
+    | false -> ordmap
+    | true -> begin
+        validate ordmap;
+        let ordmap' = test n (succ i) (insert_hlt ~k:i ~v:(i * 100) ordmap) in
+        let k, v = choose_hlt ordmap' in
+        assert (k * 100 = v);
+        let ordmap'' = remove_hlt k ordmap' in
+        validate ordmap'';
+        assert ((length ordmap') = (length ordmap'') + 1);
+        ordmap''
+      end
+  end in
+  let e = empty (module Usize) in
+  let _ = test 100 0 e in
+  printf "@]";
+
+  [%expect{|
+    |}]
```

Comment:
Unlike the `Map` and `Set` tests, this one does happen deterministically. Right? Would it be valuable to print stuff as we go?

Reply:
I sat on this pull request for a couple days, contemplating whether to add explicit output to this test.  My current thought is that it isn't necessary.  `choose` only promises to return an arbitrary binding, and this test is good enough to demonstrate that this happens.  The current implementation actually provides the stronger guarantee that the arbitrary choice is stable for equal maps (which is part of OCaml's documented semantics), but I'm not convinced that's an important guarantee.  We may want to return the root in the future, and this test will still work correctly with that change.